spec/declaration.dd

```diff
@@ -604,18 +604,34 @@ $(GNAME VoidInitializer):
         $(P Normally, variables are initialized either with an explicit
         $(GLINK Initializer) or are set to the default value for the
         type of the variable. If the $(I Initializer) is $(D void),
-        however, the variable is not initialized. If its value is
-        used before it is set, undefined program behavior will result.
+        however, the variable is not initialized.
         )
 
-        $(UNDEFINED_BEHAVIOR If a void initialized variable's value is
-        used before it is set, the behavior is undefined.
+        $(IMPLEMENTATION_DEFINED If a void initialized variable's value is
+        used before it is set, the value is implementation defined.
 
         ---
+        import std.stdio;
         void foo()
         {
             int x = void;
-            writeln(x);  // will print garbage
+            writeln(x);  // will likely print garbage
+        }
+        ---
+        )
+
+        $(UNDEFINED_BEHAVIOR If a void initialized variable is a reference type
```

Comment:
Does `reference type` wording cover types with indirections too?

Reply:
A better way to word this paragraph would be:

> Void-initialization of variables whose types have [unsafe values][1] is not allowed in `@safe` code.
>
> [1]: https://dlang.org/spec/function.html#safe-values